File: 19-remove-nth-node-from-end-of-list/19-remove-nth-node-from-end-of-list.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @param {number} n
 * @return {ListNode}
 */
var removeNthFromEnd = function(head, n) {
    
    dummy = new ListNode(0,head);
    cur = head;
    pre = dummy;
  
  
    while(n>0){
        cur = cur.next;
        n--;
    }
    while(cur){
        cur = cur.next;
        pre = pre.next;
    }
   
    pre.next = pre.next.next;
    
    return dummy.next;
    
};

File: 19-remove-nth-node-from-end-of-list/19-remove-nth-node-from-end-of-list.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */


struct ListNode* removeNthFromEnd(struct ListNode* head, int n){
    
    struct  ListNode* cur,pre,next;
    cur=head;
    int i=0;
    
    while(cur!=NULL)
    {
        cur=cur->next;
        i++;
    }

    n=i-n-1;
    printf("%d",n);
    
    cur=head;
    while (n>0){// point to the node before removal node
        cur=cur->next;
        n--;
    }
    if(n==-1)// removing first node
    {
        printf("tes\n");
        head=cur->next;
    }
      
    else
    {
        cur->next=cur->next->next;  // removing node
    }
    
  
      
    return head;
}